### PR TITLE
Adding border to table header & footer #18767

### LINKED ISSUE #18767: Table Block: Further distinguish headers/footers from regular cells
When I toggle on header/footer sections for headers, I'm expecting those rows to be more visually distinguished from regular cells, and maybe even have placeholder labels. True, the text becomes bold, but you wouldn't even know that until you click into a header cell and start typing. It needs to be more explicit, IMO.

### PR DESCRIPTION
**Pull request by dmahendrakar**
## Description
Added 2px border-bottom to header and border-top to footer. Design suggested by @shaunandrews in #18767 
image
Closes #18767.

## Screenshots 
Screen Shot 2019-12-13 at 12 31 52 PM
Screen Shot 2019-12-13 at 12 31 39 PM
Screen Shot 2019-12-13 at 12 31 11 PM
Screen Shot 2019-12-13 at 12 30 54 PM

## Types of changes
CSS changes limited to editor view.

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by dpkm95** [comment]
@ajitbohra I see there are no failing tests after after my change. Do you need me to add a snapshot test for table component? I see it's only added in block-library/embed. Is there any reason why snapshot tests are not added pervasively?

**Comment by dpkm95** [comment]
@talldan do you have any feedback on this pr?

**Comment by shaunandrews** [comment]
I like it. I wonder if it would stand out just a _little_ more at 3px?

**Comment by dpkm95** [comment]
@shaunandrews this is how it looks 2px vs 3px -
Screen Shot 2019-12-16 at 9 48 33 PM
Screen Shot 2019-12-16 at 9 48 33 PM copy

**Comment by talldan** [comment]
I tried it out with the Gutenberg Starter Theme (https://github.com/WordPress/gutenberg-starter-theme):
Screen Shot 2019-12-18 at 4 09 31 pm

At the moment the style is only applied in the editor, and I think the styles should be consistent when also viewing the post. @dpkm95 To do that, these styles could be added to the table block's theme.scss file, which should apply to both the editor and the post. That file is generally for slightly more opinionated styles that are optional for themes to use.

**Comment by dpkm95** [comment]
@talldan thanks for the feedback. Do you like 2px or 3px border? +@shaunandrews.
@talldan I'm facing an issue when I move it to style.scss[0]. In preview & render view the themes/twentytwenty/style.scss loads after block-library/table.style.scss[1], it should be other way around as it is in editor view[2]. Is this done for a reason? If not how to fix it?

[0]Screen Shot 2019-12-18 at 9 22 45 AM

[1] image

[2] Screen Shot 2019-12-18 at 9 26 27 AM

**Comment by shaunandrews** [comment]
> Do you like 2px or 3px border?

I have a slight preference for the 3px visually. Maybe others have differing opinions, I'll get more eyes.

**Comment by shaunandrews** [comment]
A few more people chimed in on Slack — lets go with 3px.

**Comment by talldan** [comment]
@dpkm95 As mentioned above, this should go in the theme.scss stylesheet, not the style.scss stylesheet.

In terms of Twenty Twenty, it was a surprise to me, but it looks like that theme doesn't support default block styles:
https://developer.wordpress.org/block-editor/developers/themes/theme-support/#default-block-styles

Any changes added to the theme.scss won't be visible when viewing a post in Twenty Twenty. It's up to the theme to add styling as it sees fit.

My recommendation would be to use a different theme to test this feature. Generally a good one to use is the Gutenberg Starter Theme (https://github.com/WordPress/gutenberg-starter-theme), as it's more of a base theme than other themes. You can download it from the repo using the `Clone or Download` button, and then upload that directly into your dev environment.

Twenty Nineteen also seems to work fine. I also tried a few older themes, but they seem to have `border-collapse: separate;` specified which stops them working correctly.

**Comment by dpkm95** [comment]
@talldan thanks for the clarification, It works for me with gutenberg-starter-theme! I've moved the styles to theme.scss, can you please review the pr.

**Comment by talldan** [comment]
@dpkm95 Looking good! What do you think about also adding the placeholder text for the header and footer cells as described in this comment - https://github.com/WordPress/gutenberg/issues/18767#issuecomment-561771034?

**Comment by dpkm95** [comment]
@talldan I've updated pr with placeholder. Thanks for the review!

**Review by talldan — APPROVED** [review]
This looks good to me, thanks for all your hard work @dpkm95!

File: packages/block-library/src/table/edit.js [hunk 1]
```diff
@@ -450,6 +450,13 @@ export class TableEdit extends Component {
 								[ `has-text-align-${ align }` ]: align,
 							}, 'wp-block-table__cell-content' );
 
+							let placeholder = '';
+							if ( name === 'head' ) {
+								placeholder = 'Header label';
+							} else if ( name === 'foot' ) {
+								placeholder = 'Footer label';
```
Comment (Soean) [inline packages/block-library/src/table/edit.js]: This string should also be translatable.

**Review by Soean — CHANGES_REQUESTED** [review]
The placeholders should be translatable. You can add the translation function like this: `__( 'placeholder label' )`

File: packages/block-library/src/table/edit.js [hunk 2]
```diff
@@ -450,6 +450,13 @@ export class TableEdit extends Component {
 								[ `has-text-align-${ align }` ]: align,
 							}, 'wp-block-table__cell-content' );
 
+							let placeholder = '';
+							if ( name === 'head' ) {
+								placeholder = 'Header label';
```
[the remark this answers is not in the document]
Comment (dpkm95) [reply packages/block-library/src/table/edit.js]: @Soean thanks

**Comment by dpkm95** [comment]
> This looks good to me, thanks for all your hard work @dpkm95!

Thanks @talldan looking forward to more contributions!

**Comment by dpkm95** [comment]
latest pr: https://github.com/WordPress/gutenberg/pull/19450 with correct github user & email

File: packages/block-library/src/table/edit.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (talldan) [reply packages/block-library/src/table/edit.js]: Good catch!

**Pull request closed without merging**